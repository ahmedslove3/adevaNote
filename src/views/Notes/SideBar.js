import React, { useState } from 'react';
// @material-ui/core components
import { makeStyles } from "@material-ui/core/styles";
import InputAdornment from "@material-ui/core/InputAdornment";
import Button from "components/CustomButtons/Button.js";
import Card from "components/Card/Card.js";
import CardBody from "components/Card/CardBody.js";
import CardFooter from "components/Card/CardFooter.js";
import CustomInput from "components/CustomInput/CustomInput.js";
import Clear from "@material-ui/icons/Clear";

// @material-ui/icons
import Search from "@material-ui/icons/Search";
import Add from "@material-ui/icons/Add";

import styles from "assets/jss/material-kit-react/views/loginPage.js";

import NoteSearchContainer from "services/noteSearch.js"
import { Subscribe } from 'unstated';

const useStyles = makeStyles(styles);



const SideBar = (props) => {
    const classes = useStyles();
    const [searchValue, setSearch] = useState('');

    const handleDelete = (e, note) => {
        e.stopPropagation();
        props.noteContainer.deleteNote(note.id);
    }

    return (

        <Card style={style.sideBar.card} >
            <CardBody>
                <CustomInput labelText="Search" id="search"

                    formControlProps={{
                        fullWidth: true,

                    }}
                    labelProps={{ style: style.searchLabel }}
                    inputProps={{
                        value: searchValue,
                        onChange: (e) => {
                            setSearch(e.target.value);

                            props.noteContainer.search(e.target.value);
                        },
                        style: style.searchInput,
                        type: "text",
                        endAdornment: (
                            <InputAdornment position="end">

                                <Search className={classes.searchIcon} />

                            </InputAdornment>
                        )
                    }}
                />
                {
                    props.noteContainer.state.notes.map(
                        (note, i) => {
                            return (
                                <div key={i} style={style.sideBar.notes} onClick={() => props.noteContainer.selectNote(note.id)} >
                                    <CardBody className=".note">
                                        <h4>{note.title}</h4>

                                        <Button onClick={(e) => handleDelete(e, note)} justIcon round style={style.deleteButton}>
                                            <Clear />
                                        </Button>

                                    </CardBody>
                                </div>
                            );
                        }
                    )
                }


                <Button onClick={() => props.noteContainer.addNote({ id: newId, title: "Note title", body: null, textBody: null, editing: true })} justIcon round style={style.addButton}>
                    <Add style={style.searchIcon} className={classes.searchIcon} />
                </Button>

            </CardBody>


        </Card>



    );
}


const style = {
    gridContainer: {
        height: "100%",
    },
    addButton: {
        backgroundColor: "#00acc1",
        position: "absolute",
        right: "20px",
        bottom: "20px",

    },
    deleteButton: {
        float: "right",
        minWidth: "10px",
        width: "10px",
        height: "10px",
        position: "absolute",
        top: "0",
        right: "0"
    },
    searchInput: {
        height: "50px",
    },
    searchLabel: {
        marginTop: "15px",
    },
    gridItem: {
        height: "100%",
    },
    cardFooter: {
        position: "relative",
    }
    ,
    sideBar: {
        card: {
            margin: '0',
            height: "100%",
            backgroundColor: "#212121"
        },
        notes: {
            margin: "0",
            width: "100%",
            color: "white",
            position: "relative"


        }

    },


}





export default SideBar;