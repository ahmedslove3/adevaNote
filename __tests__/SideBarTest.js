
import React from 'react';
import renderer from 'react-test-renderer';
import SideBar from "views/Notes/SideBar.js";
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import NoteSearchContainer from 'services/noteSearch.js';
import { Provider } from 'unstated';



test('Side bar main snapshot', () => {
    let note = new NoteSearchContainer();
    const component = renderer.create(
        <SideBar noteContainer={note} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});


