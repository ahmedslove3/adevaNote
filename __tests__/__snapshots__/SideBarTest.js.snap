// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Side bar main snapshot 1`] = `
<div
  className="makeStyles-card-11"
  style={
    Object {
      "backgroundColor": "#212121",
      "height": "100%",
      "margin": "0",
    }
  }
>
  <div
    className="makeStyles-cardBody-14"
  >
    <div
      className="MuiFormControl-root makeStyles-formControl-23 MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root makeStyles-labelRoot-20  MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink={false}
        htmlFor="search"
        style={
          Object {
            "marginTop": "15px",
          }
        }
      >
        Search
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-underline-16 MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
        onClick={[Function]}
        style={
          Object {
            "height": "50px",
          }
        }
      >
        <input
          aria-invalid={false}
          className="MuiInputBase-input MuiInput-input makeStyles-input-24 MuiInputBase-inputAdornedEnd"
          disabled={false}
          id="search"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
          value=""
        />
        <div
          className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </div>
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root makeStyles-button-95 makeStyles-round-113 makeStyles-justIcon-116 MuiButton-text"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": "#00acc1",
          "bottom": "20px",
          "position": "absolute",
          "right": "20px",
        }
      }
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;
